### PR TITLE
Enforce our Minimum Requirements for Node Operators and sanity check them

### PR DESCRIPTION
**Pull request by stefanbenten**
What: Enforces 500GB Bandwidth and Diskspace to ensure we have proper nodes on the network.

Why: Windows and OSX have issues with directory mapping sometimes, but the node still starts up.
To prevent writing to a small internal docker directory (<5GB), we check that the space is actually at least 500GB.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: storagenode/monitor/monitor.go [hunk 1]
```diff
@@ -107,6 +112,16 @@ func (service *Service) Run(ctx context.Context) (err error) {
 		service.log.Warn("Disk space is less than requested. Allocating space", zap.Int64("bytes", service.allocatedDiskSpace))
 	}
 
+	// Ensure the disk is at least 500GB in size, which is our current minimum required to be an operator
+	if service.allocatedDiskSpace < minimumDiskSpace {
+		service.log.Fatal("Total disk space less than required minimum", zap.Int64("bytes", minimumDiskSpace))
```
Comment (jtolds) [inline storagenode/monitor/monitor.go]: don't use fatal. we should return an error

Comment (stefanbenten) [reply storagenode/monitor/monitor.go]: Does that stop the node directly?

Comment (jtolds) [reply storagenode/monitor/monitor.go]: i don't know, but functions should not as a side effect call os.exit. if a service dies, we should have the service running in the peer know that certain errors are fatal errors. but it should be up to the peer code about whether to exit. having this run method decide the process should end is overstepping it's contractual bounds in my opinion

Comment (jtolds) [reply storagenode/monitor/monitor.go]: so i think if you want the node to die you should check out the peer code in the storage node and make sure that if the monitor Run method errors that it kills the storage node

Comment (jtolds) [reply storagenode/monitor/monitor.go]: maybe the error group the peer has should have two types of errors, non-fatal and fatal? and this is a fatal error.

**Comment by littleskunk** [comment]
Test looks good.
```
2019-06-09T17:22:50.603-0400	ERROR	piecestore:monitor	Total disk space less than required minimum	{"bytes": 500000000000}
Fatal error: piecestore monitor: disk space requirement not met
```

File: internal/testplanet/planet.go [hunk 2]
```diff
@@ -619,6 +620,10 @@ func (planet *Planet) newStorageNodes(count int, whitelistedSatelliteIDs []strin
 					Interval: time.Hour,
 					Timeout:  time.Hour,
 				},
+				Monitor: monitor.Config{
+					MinimumBandwidth: memory.GB * 500,
+					MinimumDiskSpace: memory.GB * 25,
```
Comment (egonelbre) [inline internal/testplanet/planet.go]: For testplanet, let's set it to 0.

Comment (egonelbre) [reply internal/testplanet/planet.go]: Or something like `100MB`.

Comment (stefanbenten) [reply internal/testplanet/planet.go]: Done

File: internal/testplanet/planet.go [hunk 3]
```diff
@@ -619,6 +620,10 @@ func (planet *Planet) newStorageNodes(count int, whitelistedSatelliteIDs []strin
 					Interval: time.Hour,
 					Timeout:  time.Hour,
 				},
+				Monitor: monitor.Config{
+					MinimumBandwidth: memory.MB * 100,
```
Comment (egonelbre) [inline internal/testplanet/planet.go]: `100 * memory.MB` usually reads better :).

**Pull request merged**